elizaOS/eliza
# fix: add required incremental option and remove invalid typescript configuration

# Relates to:
No specific issue ticket - fixing TypeScript configuration errors encountered during build

# Risks
Low - Configuration changes only, no functional code changes
- Affects TypeScript compilation settings
- No runtime behavior changes
- Only impacts development/build time performance

# Background

## What does this PR do?
Fixes two TypeScript configuration issues:
1. Adds required `incremental` option for `tsBuildInfoFile` usage
- 'tsBuildInfoFile' is used for incremental compilation but requires 'incremental: true'
- This change will improve build performance by enabling proper incremental builds

2. Removes invalid compiler option `noUncheckedSideEffectImports`
- This option is not a valid TypeScript compiler option
- Removing it resolves the TypeScript configuration error

These changes resolve the following errors:

Option 'tsBuildInfoFile' cannot be specified without specifying option 'incremental'
Unknown compiler option 'noUncheckedSideEffectImports'

## What kind of change is this?
Bug fixes (non-breaking change which fixes an issue)
- Improves TypeScript configuration
- Enables proper incremental builds
- Removes invalid configuration option

# Documentation changes needed?
My changes do not require a change to the project documentation as they only affect internal TypeScript configuration.

# Testing

## Where should a reviewer start?
Review the changes in:
1. `client/tsconfig.app.json`
2. `client/tsconfig.node.json`

## Detailed testing steps
1. Clone the repository and checkout this branch
2. Install dependencies:

```bash
cd client
pnpm install
```

4. Run TypeScript compilation to verify errors are resolved:
```bash
pnpm check-types
```

5. Verify no TypeScript configuration errors are reported
6. Run the development server to ensure everything works:
```bash
pnpm dev
```

# Deploy Notes
No special deploy steps required - changes only affect development/build configuration

- Build performance may improve due to proper incremental compilation
- No runtime changes or dependencies affected
